parser.py

```diff
@@ -1,17 +1,30 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+
 import grammar
 import visitor
 
+class PreProcessor:
+    @staticmethod
+    def remove_comments(program: str):
+        lines = program.splitlines()
+        cleaned_lines = []
+        for line in lines:
+            cleaned_line = re.sub(r'#.*$', '', line).strip()
+            if cleaned_line:
+                cleaned_lines.append(cleaned_line)
+        return "\n".join(cleaned_lines)
 
 class ImperivmParser:
     def __init__(self, grammar=grammar.imperivm, visitor=visitor.ImperivmVisitor()):
         self.grammar = grammar
         self.visitor = visitor
 
-    def parse(self, code):
-        tree = self.grammar.parse(code)
+    def parse(self, program):
+        program = PreProcessor.remove_comments(program)
```

Comment:
For better OO, we could inject the preprocessor dependency in the constructor to this class. This allows us to have different preprocessors if necessary, for instance. It also allows for easier unit testing because you can inject a testing double.

Reply:
How about something like this?
![imagen](https://github.com/user-attachments/assets/162f2d42-dc3f-4633-9bc4-423663f79862)

This way, we could have a `Preprocessor` interface with multiple implementations, depending on what we want to process, each with its own process(program) method.

P.S.: For now, I’d put it in the same file since we don’t have a folder structure, but we should start thinking about refactoring and creating one